.buildkite/pipeline.yml

```diff
@@ -7,14 +7,34 @@ steps:
     env:
       RAILS_ENV: test
     plugins:
-      - docker-compose#v3.0.0:
+      - docker-compose#v3.9.0:
           run: app
           env:
             - BUILDKITE_JOB_ID
             - CI
             - RAILS_ENV
             - BUILDKITE_ANALYTICS_TOKEN
 
+  - label: ":spider: muffet"
+    # We need to wait until rails has started before running muffet as otherwise it will error out
+    # and the test will appear to have failed without having run. The time to wait is hard to
+    # predict, and furthermore, some paths take longer to be be ready than others. The path in this
+    # loop was chosen after some non-systematic observations. So it does not guarantee that the
+    # server will be ready. But it seems to work well in practice.
+    command: 'while ! wget --spider -Sq http://app:3000/docs/agent/v3/hooks; do echo 💎🦥 Rails is still starting; sleep 0.5; done && echo 💎🏃 Rails has started running && /muffet http://app:3000/docs --include=/docs/ --exclude=https://github.com/buildkite/docs/ --max-connections=10 --color=always'
```

Comment:
This was how I've done something similar in the past before (waiting for postgres to be ready for integration specs in docker compose). Nice. It'd be worth moving it into a separate file for readability though (`.buildkite/scripts/ci-muffet` or similar - not sure of our convention yet)

Reply:
@dabarrell It's a bit tricky as `comand` here isn't run directly on the ec2 instance that the buildkite-agent is on, but rather inside a docker container that it launches through the plugin. It's possible to get a script in a docker container, but it's very cumbersome if you are not building the image yourself, as is the case here.

However, I think the readability you want is achieved by using YAML multilane strings, so I have done that now.